server: TestStatusAPIStatements failed

server.TestStatusAPIStatements [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3380769&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3380769&tab=artifacts#/) on master @ [35d2c82f8cd0d77f0487bcdbfbfd12d351293679](https://github.com/cockroachdb/cockroach/commits/35d2c82f8cd0d77f0487bcdbfbfd12d351293679):


```
                        App:                     "$ internal-update-system-locations",
                        Stats:                   roachpb.TransactionStatistics{
                            Count:      5,
                            MaxRetries: 0,
                            NumRows:    roachpb.NumericStat{Mean:1, SquaredDiffs:0},
                            ServiceLat: roachpb.NumericStat{Mean:0.0010518996, SquaredDiffs:1.6713673053199977e-08},
                            RetryLat:   roachpb.NumericStat{},
                            CommitLat:  roachpb.NumericStat{Mean:0.00031961699999999996, SquaredDiffs:2.9242306039999974e-09},
                            BytesRead:  roachpb.NumericStat{},
                            RowsRead:   roachpb.NumericStat{},
                            ExecStats:  roachpb.ExecStats{},
                        },
                        AggregatedTs: time.Time{
                            wall: 0x0,
                            ext:  63765903600,
                            loc:  (*time.Location)(nil),
                        },
                        TransactionFingerprintID: 0x78db05059f42614b,
                    },
                    NodeID: 0,
                },
                {
                    StatsData: roachpb.CollectedTransactionStatistics{
                        StatementFingerprintIDs: {0x2e180fb94d8632da},
                        App:                     "$ internal-update-system-locations",
                        Stats:                   roachpb.TransactionStatistics{
                            Count:      1,
                            MaxRetries: 0,
                            NumRows:    roachpb.NumericStat{Mean:1, SquaredDiffs:0},
                            ServiceLat: roachpb.NumericStat{Mean:0.002254526, SquaredDiffs:0},
                            RetryLat:   roachpb.NumericStat{},
                            CommitLat:  roachpb.NumericStat{Mean:2.279e-05, SquaredDiffs:0},
                            BytesRead:  roachpb.NumericStat{},
                            RowsRead:   roachpb.NumericStat{},
                            ExecStats:  roachpb.ExecStats{},
                        },
                        AggregatedTs: time.Time{
                            wall: 0x0,
                            ext:  63765903600,
                            loc:  (*time.Location)(nil),
                        },
                        TransactionFingerprintID: 0x817bb2f5cb878505,
                    },
                    NodeID: 0,
                },
            },
        }
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStatusAPIStatements658941336
--- FAIL: TestStatusAPIStatements (7.43s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestStatusAPIStatements PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server couchand
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusAPIStatements.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/status_test.go
+++ b/pkg/server/status_test.go
@@ -1813,9 +1813,17 @@ func TestStatusAPIStatements(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	params, _ := tests.CreateTestServerParams()
+	// Aug 30 2021 19:50:00 GMT+0000
+	aggregatedTs := int64(1630353000)
 	testCluster := serverutils.StartNewTestCluster(t, 3, base.TestClusterArgs{
-		ServerArgs: params,
+		ServerArgs: base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				SQLStatsKnobs: &sqlstats.TestingKnobs{
+					AOSTClause:  "AS OF SYSTEM TIME '-1us'",
+					StubTimeNow: func() time.Time { return timeutil.Unix(aggregatedTs, 0) },
+				},
+			},
+		},
 	})
 	defer testCluster.Stopper().Stop(context.Background())
 
@@ -1897,17 +1905,24 @@ func TestStatusAPIStatements(t *testing.T) {
 	// Test no params
 	testPath("statements", expectedStatements)
 	// Test combined=true forwards to CombinedStatements
-	nowInSecs := timeutil.Now().Unix()
-	testPath(fmt.Sprintf("statements?combined=true&start=%d", nowInSecs), nil)
+	testPath(fmt.Sprintf("statements?combined=true&start=%d", aggregatedTs+60), nil)
 }
 
 func TestStatusAPICombinedStatements(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	params, _ := tests.CreateTestServerParams()
+	// Aug 30 2021 19:50:00 GMT+0000
+	aggregatedTs := int64(1630353000)
 	testCluster := serverutils.StartNewTestCluster(t, 3, base.TestClusterArgs{
-		ServerArgs: params,
+		ServerArgs: base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				SQLStatsKnobs: &sqlstats.TestingKnobs{
+					AOSTClause:  "AS OF SYSTEM TIME '-1us'",
+					StubTimeNow: func() time.Time { return timeutil.Unix(aggregatedTs, 0) },
+				},
+			},
+		},
 	})
 	defer testCluster.Stopper().Stop(context.Background())
 
@@ -1989,11 +2004,13 @@ func TestStatusAPICombinedStatements(t *testing.T) {
 	// Test with no query params
 	testPath("combinedstmts", expectedStatements)
 
-	nowInSecs := timeutil.Now().Unix()
-	// Test with end = now; should give the same results as get all.
-	testPath(fmt.Sprintf("combinedstmts?end=%d", nowInSecs), expectedStatements)
-	// Test with start = 1 hour ago end = now; should give the same results as get all.
-	testPath(fmt.Sprintf("combinedstmts?start=%d&end=%d", nowInSecs-3600, nowInSecs), expectedStatements)
+	oneMinAfterAggregatedTs := aggregatedTs + 60
+	// Test with end = 1 min after aggregatedTs; should give the same results as get all.
+	testPath(fmt.Sprintf("combinedstmts?end=%d", oneMinAfterAggregatedTs), expectedStatements)
+	// Test with start = 1 hour before aggregatedTs  end = 1 min after aggregatedTs; should give same results as get all.
+	testPath(fmt.Sprintf("combinedstmts?start=%d&end=%d", aggregatedTs-3600, oneMinAfterAggregatedTs), expectedStatements)
+	// Test with start = 1 min after aggregatedTs; should give no results
+	testPath(fmt.Sprintf("combinedstmts?start=%d", oneMinAfterAggregatedTs), nil)
 }
 
 func TestListSessionsSecurity(t *testing.T) {

